src/components/dashboard/FeedItems/FeedModalItem.js

```diff
@@ -1,148 +1,151 @@
 // @flow
 import React from 'react'
-import { StyleSheet, View } from 'react-native'
 import normalize from 'react-native-elements/src/helpers/normalizeText'
+import { Image, View } from 'react-native'
 import { Avatar, BigGoodDollar, CustomButton, Text } from '../../common'
+import ModalWrapper from '../../common/modal/ModalWrapper'
 import { getFormattedDateTime } from '../../../lib/utils/FormatDate'
+import { withStyles } from '../../../lib/styles'
 import type { FeedEventProps } from './EventProps'
 import EventCounterParty from './EventCounterParty'
+import getEventSettingsByType from './EventSettingsByType'
+import EventIcon from './EventIcon'
 
 /**
  * Render modal item according to the type for feed list in horizontal view
  * @param {FeedEventProps} props - feed event
  * @returns {HTMLElement}
  */
-const FeedModalItem = ({ styles, item, onPress }: FeedEventProps) => (
-  <View style={styles ? [customStyles.horizItem, styles] : customStyles.horizItem}>
-    <View style={customStyles.fullHeight}>
-      <View style={customStyles.modal}>
-        {['send', 'empty'].indexOf(item.type) === -1 && (
-          <Text style={customStyles.dateText}>{getFormattedDateTime(item.date)}</Text>
-        )}
-        {item.data && (
-          <View style={[customStyles.row, customStyles.title]}>
-            {item.data.endpoint && item.data.endpoint.title && (
-              <Text style={customStyles.leftTitle}>{item.data.endpoint.title}</Text>
-            )}
-            <Text style={customStyles.leftTitle}>
-              {item.type === 'send' ? 'Sent G$' : 'Received G$'}
-              {item.type === 'send' && item.data.endpoint.withdrawStatus && (
-                <Text> by link - {item.data.endpoint.withdrawStatus}</Text>
-              )}
-            </Text>
-            <BigGoodDollar number={item.data.amount} elementStyles={customStyles.currency} />
+const FeedModalItem = (props: FeedEventProps) => {
+  const { item, onPress, styles, theme } = props
+  const buttonPress = () => {
+    onPress(item.id)
+  }
+  const itemType = item.type
+  const mainColor = getEventSettingsByType(theme, itemType).color
+  const getImageByType = type => {
+    return (
+      {
+        claim: require('./img/receive.png'),
+        receive: require('./img/receive.png'),
+        send: require('./img/send.png'),
+      }[type] || null
```

Comment:
:ok_hand:

Reply:
I obviously stole that pattern from you.